testsuite/cluster-test/src/main.rs

```diff
@@ -382,6 +380,13 @@ impl BasicSwarmUtil {
     }
 }
 
+impl std::ops::Deref for ClusterUtil {
```

Comment:
Deref is pretty reserved thing for wrapper types, I don't think it makes sense to have it here.
Maybe just refactor `emit_tx` method, instead of having it as an instance method on `BasicSwarmUtil`, you can make it regular function (outside of any class) that takes &Cluster as an argument?

Reply:
I tried to make BasicSwarmUtil to be the base class and ClusterUtil to be the child class, which emit_tx can be Polymorphism for both class.

But yes, I can move emit_tx to be a out side normal fn for both of them instead of what I was thinking

Reply:
Yeah, I don't think we want to next BasicSwarmUtil into ClusterUtil just for that